src/main/java/io/xpire/logic/LogicManager.java

```diff
@@ -26,24 +28,31 @@
 
     private final Model model;
     private final Storage storage;
-    private final XpireParser xpireParser;
+    private Parser parser;
+    private final XpireParser xpireParser = new XpireParser();
+    private final ReplenishParser replenishParser = new ReplenishParser();
 
     public LogicManager(Model model, Storage storage) {
         this.model = model;
         this.storage = storage;
-        this.xpireParser = new XpireParser();
+        this.parser = xpireParser;
     }
 
     @Override
     public CommandResult execute(String commandText) throws CommandException, ParseException {
         logger.info("----------------[USER COMMAND][" + commandText + "]");
 
         CommandResult commandResult;
-        Command command = this.xpireParser.parseCommand(commandText);
+        if (this.model.getCurrentFilteredItemList() == this.model.getFilteredXpireItemList()) {
```

Comment:
Are the return values primitives here? Otherwise should call .equals() method.

Reply:
Okays, will update! :)